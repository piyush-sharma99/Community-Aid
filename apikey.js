/*
 *  ClassName: apikey.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 * @reference https://www.youtube.com/watch?v=8zT6CYu0iYQ
 *
 */

export const firebaseConfig = {
  apiKey: "AIzaSyDGliQfMfKpXUllG5ZPyClJL6C8VgFKXo8",
  authDomain: "community-aid.firebaseapp.com",
  databaseURL:
    "https://community-aid-default-rtdb.europe-west1.firebasedatabase.app",
  projectId: "community-aid",
  storageBucket: "community-aid.appspot.com",
  messagingSenderId: "529896701339",
  appId: "1:529896701339:web:1c30369d39f8654b01353e",
  measurementId: "G-PPD28P0PC6",
};
