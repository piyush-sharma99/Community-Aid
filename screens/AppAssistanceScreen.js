/*
 *  ClassName: AppAssistanceSceen.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 *
 */

//Imports
import React from "react";
import {
  StyleSheet,
  Text,
  View,
  ImageBackground,
  TouchableOpacity,
} from "react-native";

//Main Component
const AppAssistanceScreen = (props) => {
  return (
    <ImageBackground source={require("../assets/BG.png")} style={styles.bImage}>
      <View style={styles.screen}>
        <Text style={styles.text}>Pick the topic of confusion:</Text>

        <TouchableOpacity
          style={styles.Btn1}
          onPress={() => {
            props.navigation.navigate({ routeName: "MapV" });
          }}
        >
          <Text style={styles.subText}>Map Assistance</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.Btn}
          onPress={() => {
            props.navigation.navigate({ routeName: "AssistanceV" });
          }}
        >
          <Text style={styles.subText}>Assistance Request Page</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.Btn}
          onPress={() => {
            props.navigation.navigate({ routeName: "ManageV" });
          }}
        >
          <Text style={styles.subText}>Manage Requests Page</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.Btn}
          onPress={() => {
            props.navigation.navigate({ routeName: "SettingV" });
          }}
        >
          <Text style={styles.subText}>Settings</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.Btn}
          onPress={() => {
            props.navigation.navigate({ routeName: "ForgotV" });
          }}
        >
          <Text style={styles.subText}>Forgot Password</Text>
        </TouchableOpacity>
      </View>
    </ImageBackground>
  );
};

//Navigation options: changing header displayed on the page
AppAssistanceScreen.navigationOptions = {
  headerTitle: "Help ",
  headerStyle: {
    backgroundColor: "#2c8ffa",
  },
  headerTintColor: "white",
};

//CSS
const styles = StyleSheet.create({
  screen: {
    flex: 1,
  },
  bImage: {
    width: "100%",
    height: "100%",
  },
  text: {
    padding: 10,
    marginRight: 40,
    marginLeft: 40,
    color: "white",
    fontWeight: "bold",
    fontSize: 25,
    textAlign: "center",
    marginTop: 40,
    borderWidth: 3,
    borderColor: "#fb5b5a",
    borderRadius: 15,
  },

  Btn1: {
    backgroundColor: "#fb5b5a",
    borderRadius: 10,
    height: 50,
    marginRight: 40,
    marginLeft: 40,
    alignItems: "center",
    justifyContent: "center",
    marginTop: 100,
    marginBottom: 5,
    borderWidth: 2,
    borderColor: "white",
    shadowOpacity: 2,
    elevation: 5,
    shadowOffset: {
      width: 3,
      height: 3,
    },
  },
  Btn: {
    backgroundColor: "#fb5b5a",
    borderRadius: 10,
    height: 50,
    marginRight: 40,
    marginLeft: 40,
    alignItems: "center",
    justifyContent: "center",
    marginTop: 30,
    marginBottom: 5,
    borderWidth: 2,
    borderColor: "white",
    shadowOpacity: 2,
    elevation: 5,
    shadowOffset: {
      width: 3,
      height: 3,
    },
  },

  subText: {
    color: "white",
    fontSize: 20,
  },
});

export default AppAssistanceScreen;
