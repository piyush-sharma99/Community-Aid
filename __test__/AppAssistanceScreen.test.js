import React from "react";
import AppAssistanceScreen from "../screens/AppAssistanceScreen";
import { firebaseConfig } from "../apikey";
import { cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

afterEach(cleanup);

test("AppAssistanceScreen renders correctly", () => {
  const tree = renderer.create(<AppAssistanceScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
