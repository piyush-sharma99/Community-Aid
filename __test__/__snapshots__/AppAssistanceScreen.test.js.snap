// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppAssistanceScreen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fb5b5a",
          "borderRadius": 15,
          "borderWidth": 3,
          "color": "white",
          "fontSize": 25,
          "fontWeight": "bold",
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 40,
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      Pick the topic of confusion:
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fb5b5a",
          "borderColor": "white",
          "borderRadius": 10,
          "borderWidth": 2,
          "elevation": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 100,
          "opacity": 1,
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Map Assistance
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fb5b5a",
          "borderColor": "white",
          "borderRadius": 10,
          "borderWidth": 2,
          "elevation": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 30,
          "opacity": 1,
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Assistance Request Page
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fb5b5a",
          "borderColor": "white",
          "borderRadius": 10,
          "borderWidth": 2,
          "elevation": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 30,
          "opacity": 1,
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Manage Requests Page
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fb5b5a",
          "borderColor": "white",
          "borderRadius": 10,
          "borderWidth": 2,
          "elevation": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 30,
          "opacity": 1,
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Settings
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fb5b5a",
          "borderColor": "white",
          "borderRadius": 10,
          "borderWidth": 2,
          "elevation": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 30,
          "opacity": 1,
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Forgot Password
      </Text>
    </View>
  </View>
</View>
`;
